### PR TITLE
CHG: Padding interface now FT compatible

### PR DESCRIPTION
**Pull request by tensionhead**
- in FT the keyword is simply `pad`, and three modes are supported:
'maxperlen' (default, do nothing or pad to longest trial.. we had `pad_to_length=None`
to achieve the same thing before)
float - padding length in seconds (we had in samples before)
'nextpow2' - nothing changed

On branch FT-padding
Changes to be committed:
	modified:   syncopy/nwanalysis/connectivity_analysis.py
	modified:   syncopy/shared/const_def.py
	modified:   syncopy/shared/input_processors.py
	modified:   syncopy/specest/freqanalysis.py
	modified:   syncopy/tests/helpers.py
	modified:   syncopy/tests/test_connectivity.py
	modified:   syncopy/tests/test_specest.py

Author Guidelines
-----------------
- [x] Is the change set **< 400 lines**?
- [x] Was the code checked for memory leaks/performance bottlenecks?
- [x] Is the code running locally **and** on the ESI cluster?
- [x] Is the code running on all supported platforms?

Reviewer Checklist
------------------
- [ ] Are testing routines present?
- [ ] Do **parallel** loops have a set length and correct termination conditions?
- [ ] Do objects in the global package namespace perform proper parsing of their input? 
- [ ] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [ ] Code layout
  - [ ] Is the code PEP8 compliant?
  - [ ] Does the code adhere to agreed-upon naming conventions?
  - [ ] Are keywords clearly named and easy to understand?
  - [ ] No commented-out code?
- [ ] Are all docstrings complete and accurate?

### CODE REVIEW

File: syncopy/specest/freqanalysis.py [hunk 1]
```diff
@@ -145,15 +145,15 @@ def freqanalysis(data, method='mtmfft', output='pow',
         but may be unbounded (e.g., ``[-np.inf, 60.5]`` is valid). Edges `fmin`
         and `fmax` are included in the selection. If `foilim` is `None` or
         ``foilim = "all"``, all frequencies are selected.
-    pad_to_length : int, None or 'nextpow2'
+    pad : 'maxperlen', float or 'nextpow2'
```
Comment (pantaray) [inline syncopy/specest/freqanalysis.py:148]: Same here - took the liberty and rephrased based on the docstring of `process_padding`

File: syncopy/nwanalysis/connectivity_analysis.py [hunk 2]
```diff
@@ -101,17 +101,15 @@ def connectivityanalysis(data, method="coh", keeptrials=False, output="abs",
...
-        to this absolute length. E.g. `pad_to_length=2000` pads all
-        trials to 2000 samples, if and only if the longest trial is
-        at maximum 2000 samples.
-
-        Alternatively if all trials have the same initial lengths
-        setting `pad_to_length='nextpow2'` pads all trials to
-        the next power of two.
-        If `None` and trials have unequal lengths all trials are padded to match
-        the longest trial.
+    pad : 'maxperlen', float or 'nextpow2'
```
Comment (pantaray) [inline syncopy/nwanalysis/connectivity_analysis.py:104]: This reads a little bumpy imo - I think the more "technical" phrasing in the docstring of process_padding makes the point a little better. I rephrased the paragraph based on it (PR commit incoming)

File: syncopy/tests/helpers.py [hunk 3]
```diff
@@ -19,30 +19,30 @@
...
-    pad_options = [pad_length, 'nextpow2', None]
+    pad_options = [pad_length, 'nextpow2', 'maxperlen']
     for pad in pad_options:
-        method_call(pad_to_length=pad)
+        method_call(pad=pad)
 
     # test invalid pads
     try:
-        method_call(pad_to_length=2)
+        method_call(pad=0.1) # trials should be longer than 0.1 seconds
```
Comment (pantaray) [inline syncopy/tests/helpers.py:31]: I suggest changing this to `pad = -0.1` or something else. Trials of 0.1 seconds length aren't completely outrageous.

**Review by pantaray — COMMENTED** [review]
Looks good - thanks for refactoring! Minor nitpick in the tests and some edits to the docstring, other than that I think we're good to merge.

File: syncopy/specest/freqanalysis.py (hunk 1 above)
Comment (tensionhead) [reply syncopy/specest/freqanalysis.py:148]: thanks, I clearly overlooked this.. 👍

File: syncopy/nwanalysis/connectivity_analysis.py (hunk 2 above)
Comment (tensionhead) [reply syncopy/nwanalysis/connectivity_analysis.py:104]: Yep, wasn't happy with it either 👍

File: syncopy/tests/helpers.py (hunk 3 above)
Comment (tensionhead) [reply syncopy/tests/helpers.py:31]: Hehe, that's in the eye of the beholder.. but jokes aside, agreed !

**Pull request merged**